src/Avalonia.Controls/TextBox.cs

```diff
@@ -446,6 +462,44 @@ public bool CanPaste
             private set { SetAndRaise(CanPasteProperty, ref _canPaste, value); }
         }
 
+        /// <summary>
+        /// Property for determining whether undo/redo is enabled
+        /// </summary>
+        public bool IsUndoEnabled
+        {
+            get { return GetValue(IsUndoEnabledProperty); }
+            set 
+            { 
+                SetValue(IsUndoEnabledProperty, value);
+                if (value == false)
```

Comment:
Property changed handling should be done in OnPropertyChanged method override or related observers. This code won't code if user will set property with SetValue(IsUndoEnabledProperty, value) directly.
For UndoLimit it's not that important, it's direct property that always will call setter, but makes sense for consistency I would say.

Reply:
Thanks. `OnPropertyChanged` now handles `IsUndoEnabledProperty`. I also fixed the bad getter/setter for `UndoLimit`. (This goes for the other PR, too -- I won't comment there so you don't get spammed.)